src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -275,9 +280,15 @@ final case class ProcessApiServiceImpl(
       s"Retrieving EService document $documentId for EService $eServiceId and descriptor $descriptorId"
     logger.info(operationLabel)
 
-    val result: Future[EServiceDoc] = catalogManagementService
-      .getEServiceDocument(eServiceId, descriptorId, documentId)
-      .map(Converter.convertToApiEserviceDoc)
+    val result: Future[EServiceDoc] = for {
+      eServiceUuid    <- eServiceId.toFutureUUID
+      descriptorUuid  <- descriptorId.toFutureUUID
+      documentIdUuid  <- documentId.toFutureUUID
+      catalogItem     <- ReadModelQueries.getEService(eServiceUuid)(readModel)
+      _               <- catalogItem.toFuture(EServiceNotFound(eServiceId))
```

Comment:
minor enhancement, you can probably just retrieve the document, returning 404 if no results are returned
```suggestion
```

Reply:
You are....right...